src/client/service.h

```diff
@@ -49,53 +49,63 @@ namespace kovri {
 namespace client {
 
 class I2PServiceHandler;
+/// @class I2PService abstract class for service
+/// @brief base class for i2p service
 class I2PService {
  public:
+  /// @brief  constructor takes ClientDestination, defaults to null ptr
+  /// @param localDestination pointer to local destination
   explicit I2PService(
       std::shared_ptr<ClientDestination> local_destination = nullptr);
   explicit I2PService(
       kovri::core::SigningKeyType kt);
   virtual ~I2PService() { ClearHandlers(); }
-
+  /// @brief add a hander to set
+  /// @param conn I2pService pointer handler to add
   inline void AddHandler(
       std::shared_ptr<I2PServiceHandler> conn) {
     std::unique_lock<std::mutex> l(m_HandlersMutex);
     m_Handlers.insert(conn);
   }
-
+  /// @brief remove handler from set
+  /// @param conn I2PServiceHandler pointer to remove
   inline void RemoveHandler(
       std::shared_ptr<I2PServiceHandler> conn) {
     std::unique_lock<std::mutex> l(m_HandlersMutex);
     m_Handlers.erase(conn);
   }
-
+  /// @brief  clear out handlers from set
   inline void ClearHandlers() {
     std::unique_lock<std::mutex> l(m_HandlersMutex);
     m_Handlers.clear();
   }
-
+  /// @brief returns pointer to member local destination
   inline std::shared_ptr<ClientDestination> GetLocalDestination() {
     return m_LocalDestination;
   }
-
+  /// @brief Set new member local destination
+  /// @param dest pointer of type ClientDestination
   inline void SetLocalDestination(
       std::shared_ptr<ClientDestination> dest) {
     m_LocalDestination = dest;
   }
-
+  /// @brief create a stream to a destination
+  /// @param stream_request_complete
+  /// @param port send port
   void CreateStream(
       StreamRequestComplete stream_request_complete,
       const std::string& dest,
       int port = 0);
-
+  /// @brief return io_service reference of member local destination
   inline boost::asio::io_service& GetService() {
     return m_LocalDestination->GetService();
   }
-
+  /// @brief virtual start service
```

Comment:
We can plainly see that this is an interface but, if the documentation were elaborate as to *why* this must be interfaced, that would be more helpful. In actuality, I don't believe we need this interface (but that's beyond the scope of this task).

If you don't want to spend more time in this area then that's fine; we'll most likely refactor the code anyway (but this would be a good learning experience).